packages/core/src/browser/preferences/preference-service.ts

```diff
@@ -84,8 +84,6 @@ export class PreferenceServiceImpl implements PreferenceService, FrontendApplica
     @postConstruct()
     protected init(): void {
         this.toDispose.push(Disposable.create(() => this._ready.reject()));
-        this.providers.push(this.schema);
```

Comment:
it is breaking change. It should be here to make sure that default values are always available, please revert.

Reply:
This part is the key to fixing the dependency loop that would arise in the following patch.

Parsing the preferences in the constructor of `PreferenceServiceImpl` requires fetching all the `PreferenceContribution`s, which means instantiating `LanguagesFrontendContribution`.  Instantiating `LanguagesFrontendContribution` requires instantiating `FrontendApplication`, which requires instantiating `PreferenceServiceImpl`.  That causes an endless loop.

Can you please suggest an alternative to break that loop?

Reply:
I've added a commit to fix it: https://github.com/theia-ide/theia/pull/3620/commits/082d39fb03eacde282b01cef311ab06b77f7813c